pyfr/plugins/tavg.py

```diff
@@ -351,3 +357,192 @@ def __call__(self, intg):
+        # Prepare the substitution dictionary
+        subs = dict(zip(self.anames, avars))
+
+        # Evaluate the function and return
+        return np.array([npeval(v, subs) for v in self.fexprs])
+
+    def _init_tavg_merge(self, file, stats, cfg):
+        self.stats, self.cfg = stats, cfg
+        self.cfgsect = cfgsect = stats.get('tavg', 'cfg-section')
+
+        self.uuid = file['mesh_uuid']
+
+        self.fields = fields = stats.get('data', 'fields').split(',')
+        self.region = cfg.get(cfgsect, 'region')
+
+        self.dkeys = [k for k in file if k.startswith('tavg_')]
+
+        # Truth table to extract collumns later
+        self.aidx = aidx = [f.startswith('avg-') for f in fields]
+        self.sidx = sidx = [f.startswith('std-') for f in fields]
+        self.anames = [f.removeprefix('avg-') for f in compress(fields, aidx)]
+        self.snames = [f.removeprefix('std-') for f in compress(fields, sidx)]
+        nsnames = len(self.snames)
+
+        # Build fun avg expressions
+        c = cfg.items_as('constants', float)
+        self.fexprs = fexprs = []
+        for k in cfg.items(cfgsect, prefix='fun-avg-'):
+            fexprs.append(cfg.getexpr(cfgsect, k, subs=c))
+
+        # Initialise std max and avg registers
+        if self.std_all:
+            self.std_max = np.zeros(nsnames)
+            self.std_avg = np.zeros(nsnames)
+
+    def _merge_data(self, out, active, avg_time):
+        a = active
+        avg_time_inv = 1/avg_time
+        t_rng0, (file0, stats0, cfg0), std_mode = a.pop(0)
+        dt0 = t_rng0.len
+
+        for key in self.dkeys:
+            avg_acc = file0[key][:, self.aidx]*dt0
+            if self.std_all:
+                var_acc = file0[key][:, self.sidx]**2*dt0
+
+            for t_rng, (file, stats, cfg), std_mode in a:
+                dt = t_rng.len
+                avg_acc += file[key][:, self.aidx]*dt
+                if self.std_all:
+                    var_acc += file[key][:, self.sidx]**2*dt
+
+            if self.fexprs:
+                avg = avg_acc.transpose(1, 0, 2)*avg_time_inv
+                fun_avg = self._eval_fun_exprs(avg)
+                out[key] = np.vstack((avg, fun_avg)).transpose(1, 0, 2)
+            else:
+                out[key] = avg_acc*avg_time_inv
+
+            if self.std_all:
+                std = np.sqrt(var_acc*avg_time_inv).transpose(1, 0, 2)
+                np.maximum(self.std_max, np.amax(std, axis=(1, 2)),
+                           out=self.std_max)
+                self.std_avg += np.mean(std, axis=(1, 2))
+
+    def _merge_meta(self, out, merged_times):
+        t_window = merge_intervals(merged_times)
+
+        cfgsect = self.stats.get('tavg', 'cfg-section')
+        self.stats.set('tavg', 'merged-from', merged_times)
+        self.cfg.set(cfgsect, 'range', t_window)
+
+        # If all files have full std stats then these can be writen
+        if self.std_all:
+            self.std_avg /= len(self.dkeys)
```

Comment:
Why are we dividing through by the number of time average quantities (which is what I think `len(self.dkeys)` corresponds to).

Reply:
Yep, that should be the number of active windows.